### DISCUSSION TITLE
Jailmaker advantage over truenas apps

### DISCUSSION THREAD

**Discussion by hros** [Jip-Hop/jailmaker#248] [General]
I understand that jailmaker enables using any app in a docker and has the ability to set any configuration
But, for the case when a truenas app is available and the app's settings are sufficient for configuration, what are the advantages of using jailmaker, if any?

How does jailmaker compare on terms of ui integration for starting and stopping apos, as well as the ease of upgrades

**Comment by Lockszmith-GH** [comment]
jailmaker has zero ui integration with TrueNAS.
It is not competing with it.

It has provided an ability to persisst system level services on a pool and run them very lean, so that something like docker can co-exist on TrueNAS SCALE that doesn't have docker.

Future TrueNAS SCALE will actually have docker based apps, and then it's just a choice of whether to run something in a jail - completely disconnected from the the TrueNAS SCALE echosystem and in your control vs relying on the TrueNAS SCALE UI.
